qemu/tests/nic_teaming.py

```diff
@@ -0,0 +1,159 @@
+        logging.info("ping pass with loss raito:%s, that less than 5" % ratio)
+
+    def team_if_exist():
+        """ judge if team is alive well."""
+
+        pid_team = session_serial.cmd_output_safe("pidof teamd")
+        pid = re.findall(r"(\d+)", pid_team)
+        str_noteam = 'Device "%s" does not exist.' % team_if
+        str_real = session_serial.cmd_output_safe("ip link show %s" % team_if)
+        if (pid == []) and (str_noteam in str_real):
+            return False
+        elif (pid != []) and (str_noteam not in str_real):
+            return True
+        else:
+            mes = 'team driver hit problem: only one of the daemon and '
+            mes += 'interface exist.'
+            raise error.TestFail(mes)
+
+    error.context("Step1: Configure the team environment", logging.info)
+    # steps of building the teaming environment starts
+    modprobe_cmd = "modprobe team"
+    session_serial.cmd_output_safe(modprobe_cmd)
+    session_serial.cmd_output_safe(params.get("createteam_cmd") % team_if)
+    # this cmd is to create the team0 and correspoding userspace daemon
+
+    if team_if_exist() == False:
+        raise error.TestFail("Interface %s is not created." % team_if)
+    # check if team0 is created successfully
+    logging.info(team_port_add())
+    logging.info(session_serial.cmd_output_safe("route -n"))
+    # this is not this case checkpoint, just to check if route works fine
+    time.sleep(10)
+    # steps of building finished
+
+    try:
+        error.context("Login in guest via ssh", logging.info)
+        # steps of testing this case starts
+        session = vm.wait_for_login(timeout=timeout)
+        dest = utils_net.get_ip_address_by_interface(params.get("netdst"))
+        count = params.get("count")
+        error.context("Step2: Check if guest can ping out:", logging.info)
+        status, output = utils_test.ping(dest=dest, count=10,
+                                         interface=team_if,
+                                         timeout=30,
+                                         session=session)
+        check_ping(status, output)
+        # small ping check if the team0 works w/o failover
+        error.context("Step3: Start failover testing until ping finished",
+                      logging.info)
+        failover_thread = utils.InterruptedThread(failover, (ifnames, ))
+        failover_thread.start()
+        # start failover loop until ping finished
+        error.context("Step4: Start ping host for %s counts"
+                      % count, logging.info)
+        try:
+            if failover_thread.is_alive():
+                status, output = utils_test.ping(dest=dest, count=count,
+                                                 interface=team_if,
+                                                 timeout=float(count) * 1.5,
+                                                 session=session)
+            else:
+                raise logging.info("The failover thread is not alive")
```

Comment:
raise be a exception not a function. eg  ValueError,  "failover thread not alive"